### PR TITLE
Publishing a 'coherent' badge; part 2

### PR DESCRIPTION
**Pull request by ghost**
Publish a 'coherent' badge to be consumed by the CLI README.

The 'Coherent' property needs to be available for both the 'FinishBuild' and 'PublishArtifacts' targets.

@dotnet/dotnet-cli

### CODE REVIEW

File: build/Publish.targets [hunk 1]
```diff
@@ -5,12 +5,14 @@
...
   <Import Project="$(MSBuildThisFileDirectory)/publish/Checksum.targets" />
-
+  <Import Project="$(MSBuildThisFileDirectory)/publish/RuntimeCoherence.targets" />
+  
   <!-- PUBLISH_TO_AZURE_BLOB env variable set by CI -->
   <Target Name="Publish"
           Condition=" '$(PUBLISH_TO_AZURE_BLOB)' != '' "
           DependsOnTargets="Init;
                             Package;
+                            RuntimeCoherence;
```
Comment (wli3) [inline build/Publish.targets]: nit pick: all task start with verb, maybe this should be _evaluateRuntimeCoherence_

File: build/publish/RuntimeCoherence.targets [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      <File Include="$(AspNetCoreSharedRuntimeVersionFile)" />
+    </ItemGroup>
+    <ReadLinesFromFile File="@(File)">
+      <Output
+          TaskParameter="Lines"
+          ItemName="AspNetCoreSharedRuntimeVersion"/>
+    </ReadLinesFromFile>
+  </Target>
+  
+  <Target Name="EvaluateRuntimeCoherence"
```
Comment (wli3) [inline build/publish/RuntimeCoherence.targets]: nit pick: and this called compareVersionNumbers?

File: build/publish/RuntimeCoherence.targets [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Target Name="RuntimeCoherence"
+          DependsOnTargets="ReadAspNetCoreSharedRuntimeVersionFile;
```
Comment (wli3) [inline build/publish/RuntimeCoherence.targets]: I don't know if this is redundant, the following should covered it already. However I am not sure this is a common practice in build script
```xml
 <Target Name="EvaluateRuntimeCoherence"
DependsOnTargets="ReadAspNetCoreSharedRuntimeVersionFile">
```

**Review by wli3 — APPROVED** [review]

**Review by livarcocc — APPROVED** [review]

**Comment by livarcocc** [comment]
@MattGertz Infra-structure change only.

File: build/Publish.targets (hunk 1 above)
Comment (johnbeisner) [inline build/Publish.targets]: done...

File: build/publish/RuntimeCoherence.targets (hunk 3 above)
Comment (johnbeisner) [inline build/publish/RuntimeCoherence.targets]: done...

File: build/publish/RuntimeCoherence.targets (hunk 2 above)
Comment (johnbeisner) [inline build/publish/RuntimeCoherence.targets]: done...

**Pull request merged**